src/sentry/integrations/bitbucket/search.py

```diff
@@ -1,73 +1,67 @@
-            return Response(status=404)
-
-        field = request.GET.get("field")
-        query = request.GET.get("query")
-        if field is None:
-            return Response({"detail": "field is a required parameter"}, status=400)
-        if not query:
-            return Response({"detail": "query is a required parameter"}, status=400)
+    @property
+    def repository_field(self):
+        return "repo"
 
-        installation = integration.get_installation(organization_id=organization.id)
-        assert isinstance(installation, BitbucketIntegration), installation
+    @property
+    def integration_provider(self):
+        return "bitbucket"
 
-        if field == "externalIssue":
-            repo = request.GET.get("repo")
-            if not repo:
-                return Response({"detail": "repo is a required parameter"}, status=400)
+    @property
+    def installation_class(self):
+        return BitbucketIntegration
 
-            full_query = f'title~"{query}"'
-            try:
-                resp = installation.search_issues(query=full_query, repo=repo)
-            except ApiError as e:
-                if "no issue tracker" in str(e):
-                    logger.info(
-                        "bitbucket.issue-search-no-issue-tracker",
-                        extra={"installation_id": installation.model.id, "repo": repo},
-                    )
-                    return Response(
-                        {"detail": "Bitbucket Repository has no issue tracker."}, status=400
-                    )
-                raise
-            return Response(
-                [
-                    {"label": "#{} {}".format(i["id"], i["title"]), "value": i["id"]}
-                    for i in resp.get("values", [])
-                ]
-            )
+    def handle_search_issues(self, installation: T, query: str, repo: str) -> Response:
+        full_query = f'title~"{query}"'
+        try:
+            response = installation.search_issues(query=full_query, repo=repo)
+        except ApiError as e:
+            if "no issue tracker" in str(e):
+                logger.info(
+                    "bitbucket.issue-search-no-issue-tracker",
+                    extra={"installation_id": installation.model.id, "repo": repo},
+                )
+                return Response(
+                    {"detail": "Bitbucket Repository has no issue tracker."}, status=400
+                )
+            raise
 
-        if field == "repo":
-            result = installation.get_repositories(query)
-            return Response([{"label": i["name"], "value": i["name"]} for i in result])
+        assert isinstance(response, dict)
+        return Response(
+            [
+                {"label": "#{} {}".format(i["id"], i["title"]), "value": i["id"]}
+                for i in response.get("values", [])
+            ]
+        )
 
-        return Response(status=400)
+    # TODO: somehow type installation with installation_class
```

Comment:
Do you still need this comment, or did the generic change cover it?

Reply:
oopsie